src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div
+    v-if="!isBadge"
+    class="k-copy"
+    :class="truncate ? null : 'copy-truncated'"
+  >
+    <KTooltip
+      v-if="format !== 'hidden'"
+      :class="truncate ? null : 'copy-content'"
+      data-testid="copy-content"
+      :label="contentTooltip"
+      placement="bottomStart"
+      position-fixed
+    >
+      <div
+        :class="[
+          'content-container',
+          truncate ? 'truncated-content' : 'non-truncated-content',
+          monospace ? 'monospace' : null
+        ]"
+        :style="widthStyle"
+      >
+        {{ contentFormat }}
+      </div>
+    </KTooltip>
+
+    <KTooltip
+      class="content-icon-wrapper"
+      :label="`${props.badgeText}` ? iconTitle : tooltipText"
+      max-width="500px"
+      placement="bottomStart"
+      position-fixed
+    >
+      <KClipboardProvider v-slot="{ copyToClipboard }">
+        <span
+          data-testid="copy-to-clipboard"
+          role="button"
+          tabindex="0"
+          @click.stop="copyIdToClipboard(copyToClipboard)"
+        >
+          <KIcon
+            class="content-icon"
+            :hide-title="!!copyTooltip || undefined"
+            icon="copy"
+            size="16"
+            :title="iconTitle"
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+
+  <div
+    v-else
+    class="k-copy-badge"
+  >
+    <span
+      v-if="isBadge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <KBadge
+      v-if="isBadge"
+      appearance="custom"
+      background-color="#f9fafb"
+      border-color="#d7d8fe"
+      class="id-badge"
+      color="#473cfb"
+      is-bordered
+      shape="rectangular"
+    >
+      <KCopy
+        :badge-text="badgeText"
+        class="id-badge-content"
+        :content="content"
+        :content-tooltip="contentTooltip"
+        :copy-tooltip="copyTooltip"
+        :monospace="monospace"
+        :success-tooltip="successTooltip"
+        :truncate="truncate"
+        :truncation-limit="truncationLimit"
+      />
+    </KBadge>
```

Comment:
question: maybe I'm thinking about this wrong, but if the `KCopy` element has a badge appearance, why are we recursively nesting another `KCopy` element within a `KBadge` to provide the copy functionality?

Reply:
Would a v-if/v-else make more sense here instead of the recursive nesting? I was trying not to have the same code repeated under both the blocks of v-if/v-else.